cardano-node/src/Cardano/Tracing/Startup.hs

```diff
@@ -0,0 +1,273 @@
+    toObject _verb (StartupTime time) =
+      mkObject [ "startupTime" .= String ( Text.pack
+                                         . show
+                                         . (ceiling :: POSIXTime -> Int)
+                                         . utcTimeToPOSIXSeconds
+                                         $ time
+                                         )
+               ]
+    toObject _verb (StartupNetworkMagic networkMagic) =
+      mkObject [ "networkMagic" .= String ( Text.pack . show . unNetworkMagic
+                                          $ networkMagic) ]
+    toObject _verb (StartupSocketConfigError err) =
+      mkObject [ "error" .= String (Text.pack . show $ err) ]
+    toObject _verb StartupDBValidation =
+      mkObject [ "message" .= String "start db validation" ]
+    toObject _verb NetworkConfigUpdate =
+      mkObject [ "message" .= String "ntework configuration update" ]
+    toObject _verb (NetworkConfigUpdateError err) =
+      mkObject [ "error" .= String err ]
+    toObject _verb (NetworkConfig localRoots publicRoots useLedgerAfter) =
+      mkObject [ "localRoots" .= Aeson.toJSON localRoots
+               , "publicRoots" .= Aeson.toJSON publicRoots
+               , "useLedgerAfter" .= UseLedger useLedgerAfter
+               ]
+    toObject _verb P2PWarning = 
+      mkObject [ "message" .= String p2pWarningMessage ]
+    toObject _verb P2PWarningDevelopementNetworkProtocols =
+      mkObject [ "message" .= String p2pWarningDevelopmentNetworkProtocolsMessage ]
+    toObject _ver (WarningDevelopmentNetworkProtocols ntnVersions ntcVersions) =
+      mkObject [ "message" .= String "enabled development network protocols"
+               , "nodeToNodeDevelopmentVersions" .= String (Text.pack . show $ ntnVersions)
+               , "nodeToClientDevelopmentVersions" .= String (Text.pack . show $ ntcVersions)
+               ]
+
+
+
+-- | Pretty print 'StartupInfoTrace'
+--
+ppStartupInfoTrace :: ( Show (BlockNodeToNodeVersion blk)
+                      , Show (BlockNodeToClientVersion blk)
+                      )
+                   => StartupTrace blk
+                   -> Text
+ppStartupInfoTrace (StartupInfo addresses
+                                localSocket
+                                supportedNodeToNodeVersions
+                                supportedNodeToCleintVersions)
+  = Text.pack
+  $ "\n" ++ intercalate "\n"
+    [ "node addresses:          " ++ intercalate ", " (map ppN2NSocketInfo addresses)
+    , "local socket:            " ++ maybe "NONE" ppN2CSocketInfo localSocket
+    , "node-to-node versions:\n"
+       ++ intercalate "\n"
+          (map (\(v, bv) -> show v ++ "\t" ++ show bv)
+        . Map.assocs
+        $ supportedNodeToNodeVersions)
+    , "node-to-client versions:\n"
+       ++ intercalate "\n"
+          (map (\(v, bv) -> show v ++ "\t" ++ show bv)
+        . Map.assocs
+        $ supportedNodeToCleintVersions)
+    ]
```

Comment:
Seems like a lot of string appends.  Do we need to worry about performance?

Reply:
No, not at all; this runs once at startup.